src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,634 @@
+		$has_dependencies       = ! empty( $this->get_plugin_dependencies( $plugin_file ) );
+		$in_circular_dependency = $this->in_circular_dependency( $plugin_file );
+
+		// Remove deactivation link from dependencies.
+		if ( ! empty( $this->dependencies_parents[ $plugin_file ] ) && ! $in_circular_dependency ) {
+			unset( $actions['deactivate'] );
+		}
+
+		// On plugins with unmet dependencies that the user has already requested for the plugin's activation,
+		// removes the activation link from its actions and add a "Cancel pending activation" link in its place.
+		if ( $pending_activation && $has_dependencies ) {
+			unset( $actions['activate'] );
+			if ( current_user_can( 'activate_plugin', $plugin_file ) ) {
+				$cancel_activation = sprintf(
+					'<a href="%s" class="cancel-activate unmet-dependencies" aria-label="%s">%s</a>',
+					wp_nonce_url( 'plugins.php?action=cancel-activate&amp;plugin=' . rawurlencode( $plugin_file ), 'cancel-activate-plugin_' . $plugin_file ),
+					/* translators: %s: Plugin name. */
+					esc_attr( sprintf( _x( 'Cancel activation of %s', 'plugin' ), get_plugin_data( WP_PLUGIN_DIR . '/' . $plugin_file )['Name'] ) ),
+					__( 'Cancel activation request' )
+				);
+
+				$actions = array_merge( array( 'cancel-activation' => $cancel_activation ), $actions );
+			}
+		}
+		return $actions;
+	}
+
+	/**
+	 * Add dependencies info in plugins.
+	 *
+	 * @since 5.9.0
+	 * @access public
+	 *
+	 * @param string $plugin_file Path to the plugin file relative to the plugins directory.
+	 * @param array  $plugin_data An array of plugin data.
+	 *
+	 * @return void
+	 */
+	public function after_plugin_row( $plugin_file, $plugin_data ) {
+
+		$pending_activation     = in_array( $plugin_file, $this->get_plugins_to_activate(), true );
+		$dependencies           = $this->get_plugin_dependencies( $plugin_file );
+		$has_dependencies       = ! empty( $dependencies );
+		$in_circular_dependency = $this->in_circular_dependency( $plugin_file );
+		$is_plugin_active       = is_plugin_active( $plugin_file );
+
+		// Add extra info to dependencies.
+		if ( ! empty( $this->dependencies_parents[ $plugin_file ] ) ) {
+			$parents_names = array();
+			foreach ( $this->dependencies_parents[ $plugin_file ] as $parent ) {
+				$parents_names[] = get_plugin_data( WP_PLUGIN_DIR . '/' . $parent )['Name'];
+			}
+
+			$notice_contents = ( 1 < count( $parents_names ) )
+				? sprintf(
+					/* translators: %1$s: plugin name. %2$s: Parent plugin names, comma-separated. */
+					esc_html__( 'Plugin %1$s is a dependency for the following plugins: %2$s.' ),
+					esc_html( $plugin_data['Name'] ),
+					esc_html( implode( ', ', $parents_names ) )
+				) : sprintf(
+					/* translators: %1$s: plugin name. %2$s: Parent plugin name. */
+					esc_html__( 'Plugin %1$s is a dependency for the "%2$s" plugin.' ),
+					esc_html( $plugin_data['Name'] ),
+					esc_html( $parents_names[0] )
+				);
```

Comment:
I'd write this as

```php
$notice_contents = sprintf(
	/* translators: %1$s: plugin name. %2$s: Parent plugin name. */
	esc_html__('Plugin %1$s is a dependency for the "%2$s" plugin.'),
	esc_html($plugin_data['Name']),
	esc_html($parents_names[0])
);

if ( 1 < count( $parents_names ) ) {
	$notice_contents = sprintf(
		/* translators: %1$s: plugin name. %2$s: Parent plugin names, comma-separated. */
		esc_html__('Plugin %1$s is a dependency for the following plugins: %2$s.'),
		esc_html($plugin_data['Name']),
		esc_html(implode(', ', $parents_names))
	);
}
```

It's a bit more readable than a ternary in my opinion, and removes an unnecessary else 🙂